pkg/ingress/model_build_target_group.go

```diff
@@ -171,6 +172,21 @@ func (t *defaultModelBuildTask) buildTargetGroupTargetType(_ context.Context, sv
 	}
 }
 
+// buildTargetGroupPort constructs the TargetGroup's port.
+// Note: TargetGroup's port setting don't actually matter to this controller since we always register targets with port specified.
+// however, we'll do our best to use the most appropriate port as TargetGroup's port to avoid UX confusing.
+func (t *defaultModelBuildTask) buildTargetGroupPort(_ context.Context, targetType elbv2model.TargetType, svcPort corev1.ServicePort) int64 {
+	if targetType == elbv2model.TargetTypeInstance {
+		return int64(svcPort.NodePort)
+	}
+	if svcPort.TargetPort.Type == intstr.Int {
+		return int64(svcPort.TargetPort.IntValue())
+	}
+
+	// when a literal targetPort is used, it can actually be different ports for different pods. so we use a fixed 1 here.
+	return 1
```

Comment:
Is target group->port optional? How come this dummy port value has no impact?

Reply:
it's not in the data path. we override each target's port on registeration.